lib/src/react_client/event_helpers.dart

```diff
@@ -564,60 +736,74 @@ SyntheticWheelEvent createSyntheticWheelEvent({
+  /// Uses Duck Typing to detect if the event instance is a [SyntheticFocusEvent].
+  bool get isFocusEvent => hasProperty(this, 'relatedTarget') && !hasProperty(this, 'button');
 
-  /// Returns whether this is a [SyntheticFocusEvent].
-  bool get isFocusEvent => this is SyntheticFocusEvent;
+  /// Uses Duck Typing to detect if the event instance is a [SyntheticFormEvent].
+  ///
+  /// __NOTE:__ This getter is here for completeness, but because the interface for form events
+  /// is the same as that of a [SyntheticEvent] (the base for all other synthetic event types),
+  /// via Duck Typing every [SyntheticEvent] is considered a [SyntheticFormEvent].
+  bool get isFormEvent => true;
 
-  /// Returns whether this is a [SyntheticFormEvent].
-  bool get isFormEvent => this is SyntheticFormEvent;
+  /// Uses Duck Typing to detect if the event instance is a [SyntheticMouseEvent].
+  bool get isMouseEvent => hasProperty(this, 'button');
 
-  /// Returns whether this is a [SyntheticMouseEvent].
-  bool get isMouseEvent => this is SyntheticMouseEvent;
+  /// Uses Duck Typing to detect if the event instance is a [SyntheticPointerEvent].
+  bool get isPointerEvent => hasProperty(this, 'pointerId');
 
-  /// Returns whether this is a [SyntheticPointerEvent].
-  bool get isPointerEvent => this is SyntheticPointerEvent;
+  /// Uses Duck Typing to detect if the event instance is a [SyntheticTouchEvent].
+  bool get isTouchEvent => hasProperty(this, 'targetTouches');
 
-  /// Returns whether this is a [SyntheticTouchEvent].
-  bool get isTouchEvent => this is SyntheticTouchEvent;
+  /// Uses Duck Typing to detect if the event instance is a [SyntheticTransitionEvent].
+  bool get isTransitionEvent => hasProperty(this, 'propertyName');
 
-  /// Returns whether this is a [SyntheticTransitionEvent].
-  bool get isTransitionEvent => this is SyntheticTransitionEvent;
+  /// Uses Duck Typing to detect if the event instance is a [SyntheticAnimationEvent].
+  bool get isAnimationEvent => hasProperty(this, 'animationName');
 
-  /// Returns whether this is a [SyntheticAnimationEvent].
-  bool get isAnimationEvent => this is SyntheticAnimationEvent;
+  /// Uses Duck Typing to detect if the event instance is a [SyntheticUIEvent].
+  bool get isUiEvent => hasProperty(this, 'detail');
 
-  /// Returns whether this is a [SyntheticUIEvent].
-  bool get isUiEvent => this is SyntheticUIEvent;
+  /// Uses Duck Typing to detect if the event instance is a [SyntheticWheelEvent].
+  bool get isWheelEvent => hasProperty(this, 'deltaX');
+}
 
-  /// Returns whether this is a [SyntheticWheelEvent].
-  bool get isWheelEvent => this is SyntheticWheelEvent;
+extension DataTransferHelper on SyntheticMouseEvent {
+  /// The data that is transferred during a drag and drop interaction.
+  ///
+  /// See <https://developer.mozilla.org/en-US/docs/Web/API/DragEvent/dataTransfer>
+  SyntheticDataTransfer get dataTransfer => syntheticDataTransferFactory(getProperty(this, 'dataTransfer'));
```

Comment:
Sorry, I was the one that suggested an extension, but I just thought: can we make this a normal getter? Would you mind trying that out?

Reply:
Yep! I gave it a go and it behaves weirdly. Even though I define the getter on the interop class, which I'd expect would override the `external`, calling `event.dataTransfer` appears to access the external getter anyway. It only happens when the getter is on the class itself and not an extension though. I stepped through in the debugger and when the getter is in the interop class, it acts like a simple getter and there isn't anything to step into (it just grabs the property off the JS object). As an extension though, I can step through to the `syntheticDataTransferFactory`. 

So that's weird. Let me know if you want to step through it together and we definitely can!

Reply:
[This](https://github.com/cleandart/react-dart/pull/284#discussion_r526340434) might be a related issue as well

Reply:
Makes sense, sounds like this is just one of the unwritten rules of JS interop. Thanks for looking into it!!